### PR TITLE
(Not for submission) Prototype of fake for PagedEnumerable

### PR DESCRIPTION
**Pull request by jskeet**
Not happy with "FakeDataSource" as a title - suggestions welcome.
Obviously I'll add tests, when we're happy with the design.

(This could make the tests for GrpcPagedEnumerable etc simpler,
potentially.)

### CODE REVIEW

File: testing/Google.Api.Gax.Testing/FakeDataSource.cs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                Task.FromResult(_parent.ReadPage(_initialPageToken, pageSize));
+
+            public override IAsyncEnumerator<TResource> GetEnumerator()
+            {
+                var iterator = _parent.GetEnumerator(_initialPageToken);
+                return AsyncEnumerable.CreateEnumerator(token => Task.FromResult(iterator.MoveNext()), () => iterator.Current, iterator.Dispose);
+            }
+        }
+
+        private class FakePagedEnumerable : PagedEnumerable<TResponse, TResource>
```
Comment (chrisdunelm) [inline testing/Google.Api.Gax.Testing/FakeDataSource.cs]: Need to override GetEnumerator()

**Comment by chrisdunelm** [comment]
How about `TestPagedDataSource` ?

File: testing/Google.Api.Gax.Testing/FakeDataSource.cs (hunk 1 above)
Comment (jskeet) [reply testing/Google.Api.Gax.Testing/FakeDataSource.cs]: Oh yes - done.

**Comment by jskeet** [comment]
Think I'd prefer Fake over Test. But yes, FakePagedDataSource could work...

**Comment by sduskis** [comment]
@jskeet, this PR is old.  Can it be closed?

**Comment by jskeet** [comment]
@sdukis: It's old, but could still be useful in the future. I'll have another look today and see.

**Comment by jskeet** [comment]
Closing for now as there's not enough value here to preserve. I'll leave the branch in my fork though.

**Pull request closed without merging**